src/cosalib/metal.py

```diff
@@ -0,0 +1,316 @@
+        used. Default to building "metal" disk targets.
+        """
+        self._image_name = None
+        variant = kwargs.pop("variant", "metal")
+        if variant:
+            kwargs.update(VARIANTS.get(variant, {}))
+        self.arch = kwargs.pop("arch", get_basearch)
+        self.force = kwargs.get("force", False)
+        self.format = kwargs.pop("format")
+        self.kargs = kwargs.pop("kargs", [])
+        self.platform = kwargs.pop("platform")
+        self.target_args = kwargs.pop("target_args", [])
+
+        # Load the _Build super class
+        _Build.__init__(self, *args, **kwargs)
+        # Load the _BuildOSTree super class.
+        _BuildOSTree.__init__(self, self)
+
+        # Read in the image.yaml and manifest.yaml files
+        self.img_cfg = self.read_yaml_file("image.yaml")
+        self.manifest = self.read_yaml_file("manifest.yaml")
+
+        # sanity check the OSTree against the expected ref
+        ref = self.manifest.get("ref")
+        if ref and ref.replace("${arch}", self.basearch) != ref:
+            raise MetalImageError(f"expected ref {ref} mismatches")
+
+        # this is used in case the image has a different disk
+        # name than the platform
+        self.platform_image_name = kwargs.get(
+            "platform_image_name", self.platform)
+
+        # building dasd for ppc64el makes no sense
+        if self.arch != "any" and self.arch != self.basearch:
+            raise MetalImageError(
+                f"{self.basearch} cannot be built for disk {variant}")
+
+        # only support known-good architectures, eg. no sparks
+        if self.basearch not in ["x86_64", "aarch64", "s390x", "ppc64le"]:
+            raise MetalImageError(
+                f"{self.basearch} is not support for metal-type disk images")
+
+    def read_yaml_file(self, yaml_fname):
+        """
+        Get the contents of yaml_fname relative to the <dir>/src/config
+        """
+        fname = os.path.join(self.workdir, "src", "config", f"{yaml_fname}")
+        if not os.path.exists(fname):
+            raise Exception(f"failed to find file {fname}")
+        data = None
+        with open(fname, 'r') as f:
+            data = yaml.safe_load(f.read())
+        if not data:
+            raise Exception(f"{yaml_fname}.yaml did not read in properly")
+        return data
+
+    @property
+    def image_name(self):
+        if self._image_name is not None:
+            return self._image_name
+
+        return (f'{self.build_name}-{self.build_id}'
+                f'-{self.platform}.{self.basearch}.{self.format}')
+
+    @property
+    def rootfs_size(self):
+        """
+        Returns the rootfs size and the image size.
+        """
+        size = self.ostree_repo_estimated_size
+        return f"{int(size * 1.35)}M"
+
+    @property
+    def disk_size(self):
+        """
+        Returns the size of the disk. This is the size of the
+        rootfs with padding
```

Comment:
This is for the boot partition right?  Would be good to note that and probably hoist the `513` to a visible constant above.